src/history/hs.c

```diff
@@ -561,11 +446,17 @@ __wt_hs_insert_updates(WT_CURSOR *cursor, WT_BTREE *btree, WT_RECONCILE *r, WT_M
     if (!btree->hs_entries)
         btree->hs_entries = true;
 
-    /* Wrap all the updates in a transaction. */
-    WT_ERR(__wt_txn_begin(session, NULL));
-    __hs_set_isolation(session, &saved_isolation);
-
-    local_txn = true;
+    /*
+     * Wrap all the updates in a transaction:
+     * 1. We should already have a running application transaction, or
+     * 2. We create one.
+     * TODO: WT-5478 Ideally we should not need transaction semantics (or a txn mod structure) for a
+     * write to history store.
+     */
+    if (!F_ISSET(&session->txn, WT_TXN_RUNNING)) {
+        WT_ERR(__wt_txn_begin(session, NULL));
```

Comment:
Note: The tests prove that we *are* starting a transaction from the application session just to write to history store. I confirmed that by adding the following assert in the if condition:
WT_ASSERT(session, F_ISSET(session, WT_SESSION_INTERNAL));

This could interfere with implicit/auto application transactions. WT-5478 will investigate methods to bypass having to create a transaction itself.

@agorrod, you are not needed for a review for this pull request, but we discussed the following, so wanted you to be aware.

Reply:
Thanks @sulabhM - I think it's the right thing to do for now. Let's follow up in another ticket about what the long term solution for using a history store cursor when there is no active transaction is.